app/controllers/images_controller.rb

```diff
@@ -145,62 +131,58 @@ def update
                               end
 
         TimelineEntry.create_for_revision(entry_type: timeline_entry_type,
-                                          edition: current_edition)
+                                          edition: edition)
 
-        current_edition.assign_revision(next_revision, current_user).save!
-        PreviewService.new(current_edition).try_create_preview
+        edition.assign_revision(next_revision, current_user).save!
+        PreviewService.new(edition).try_create_preview
       end
 
       if lead_image_selected?(current_revision, next_revision)
-        redirect_to document_path(@document),
+        redirect_to document_path(edition.document),
                     notice: t("documents.show.flashes.lead_image.selected",
-                              file: @image_revision.filename)
+                              file: image_revision.filename)
       elsif lead_image_removed?(current_revision, next_revision)
-        redirect_to images_path(@document),
+        redirect_to images_path(edition.document),
                     notice: t("images.index.flashes.lead_image.removed",
-                              file: @image_revision.filename)
+                              file: image_revision.filename)
       else
-        redirect_to images_path(@document)
+        redirect_to images_path(edition.document)
       end
     end
   end
 
   def destroy
-    Document.transaction do
-      document, image_revision = find_locked_document_and_image_revision(
-        params[:document_id],
-        params[:image_id],
-      )
-
-      current_edition = document.current_edition
-      current_revision = current_edition.revision
+    Edition.find_and_lock_current(document: params[:document]) do |edition|
+      image_revision = edition.image_revisions.find_by!(image_id: params[:image_id])
+      current_revision = edition.revision
```

Comment:
Minor, but it looks like you've been aiming to say `revision` instead of `current_revision` :-).

Reply:
I'm going to go back to current_revision in the other file here for consistency, as the consequences of changing that for images is more complex.

This should be a low running concerns as your updater work clears it out.